alacritty/src/renderer/text/gles2.rs

```diff
@@ -0,0 +1,494 @@
+            add_attr!(4, gl::UNSIGNED_BYTE, u8);
+
+            // Background color.
+            add_attr!(4, gl::UNSIGNED_BYTE, u8);
+
+            // Cleanup.
+            gl::BindVertexArray(0);
+            gl::BindBuffer(gl::ELEMENT_ARRAY_BUFFER, 0);
+            gl::BindBuffer(gl::ARRAY_BUFFER, 0);
+        }
+
+        Ok(Self {
+            program,
+            vao,
+            vbo,
+            ebo,
+            atlas: vec![Atlas::new(ATLAS_SIZE)],
+            batch: Batch::new(),
+            current_atlas: 0,
+            active_tex: 0,
+        })
+    }
+
+    fn with_api<F, T>(&mut self, _: &SizeInfo, func: F) -> T
+    where
+        F: FnOnce(RenderApi<'_>) -> T,
+    {
+        unsafe {
+            gl::UseProgram(self.program.id());
+            gl::BindVertexArray(self.vao);
+            gl::BindBuffer(gl::ELEMENT_ARRAY_BUFFER, self.ebo);
+            gl::BindBuffer(gl::ARRAY_BUFFER, self.vbo);
+            gl::ActiveTexture(gl::TEXTURE0);
+        }
+
+        let res = func(RenderApi {
+            active_tex: &mut self.active_tex,
+            batch: &mut self.batch,
+            atlas: &mut self.atlas,
+            current_atlas: &mut self.current_atlas,
+            program: &mut self.program,
+        });
+
+        unsafe {
+            gl::BindBuffer(gl::ELEMENT_ARRAY_BUFFER, 0);
+            gl::BindBuffer(gl::ARRAY_BUFFER, 0);
+            gl::BindVertexArray(0);
+
+            gl::UseProgram(0);
+        }
+
+        res
+    }
+}
+
+impl Drop for Gles2Renderer {
+    fn drop(&mut self) {
+        unsafe {
+            gl::DeleteBuffers(1, &self.vbo);
+            gl::DeleteBuffers(1, &self.ebo);
+            gl::DeleteVertexArrays(1, &self.vao);
+        }
+    }
+}
+
+impl TextRenderer for Gles2Renderer {
+    fn draw_cells<I: Iterator<Item = RenderableCell>>(
+        &mut self,
+        size_info: &SizeInfo,
+        glyph_cache: &mut GlyphCache,
+        cells: I,
+    ) {
+        self.with_api(size_info, |mut api| {
+            for cell in cells {
+                api.draw_cell(cell, glyph_cache, size_info)
+            }
+        })
+    }
+
+    fn resize(&self, size_info: &SizeInfo) {
+        unsafe {
+            gl::UseProgram(self.program.id());
+            self.program.update_projection(
+                size_info.width(),
+                size_info.height(),
+                size_info.padding_x(),
+                size_info.padding_y(),
+            );
+            gl::UseProgram(0);
+        }
+    }
+
+    fn loader_api(&mut self) -> LoaderApi<'_> {
+        LoaderApi {
+            active_tex: &mut self.active_tex,
+            atlas: &mut self.atlas,
+            current_atlas: &mut self.current_atlas,
+        }
+    }
+}
+
+/// Maximum items to be drawn in a batch.
+///
+/// We use the closest dividable by 4(amount of vertices we push for a glyph) number to `u16::MAX`,
+/// since it's maximum possible index in `glDrawElements` in gles2.
```

Comment:
```suggestion
/// We use the closest number to `u16::MAX` dividable by 4 (amount of vertices we push for a glyph),
/// since it's the maximum possible index in `glDrawElements` in gles2.
```